pkg/api/metadata.go

```diff
@@ -22,11 +22,29 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"runtime/debug"
 
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
 	"github.com/getsentry/sentry-go"
 	"github.com/golang/glog"
 )
 
+// getCommitSha() returns the current version (vcs.revision attribute of BuildInfo struct) using a built-in debug package and its ReadBuildInfo() function.
+// Upon error, an empty string is returned along with a GeneralError, as well as when vsc.revision is empty/unavailable.
```

Comment:
// Upon error, an empty string is returned along with a GeneralError, as well as when vsc.revision is empty/unavailable.

This is not accurate.
When ReadBuildInfo returns successfully but  `vcs.revision` is empty/unavailable then no error is returned